go/vt/vtgate/planbuilder/dml.go

```diff
@@ -39,7 +39,17 @@ func buildUpdatePlan(upd *sqlparser.Update, vschema VSchema) (*engine.Route, err
 	route := &engine.Route{
 		Query: generateQuery(upd),
 	}
-	updateTable, _ := upd.Table.Expr.(sqlparser.TableName)
+	if len(upd.TableExprs) > 1 {
+		return nil, errors.New("unsupported: multi-table update")
+	}
+	aliased, ok := upd.TableExprs[0].(*sqlparser.AliasedTableExpr)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
+	updateTable, ok := aliased.Expr.(sqlparser.TableName)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
```

Comment:
This section will nullify the feature because it will block all multi-table updates. Instead, you could call processTableExprs, which will return a builder. Then you can verify it with:
```
if rb, ok := bldr.(*route); !ok || rb.ERoute.Keyspace.Sharded {
  return nil, errors.New("unsupported: update statement spans multiple shards")
}
```
which will allow all single shard updates to pass through.

Same should work for deletes.

You'll need to add some tests in dml_cases.txt for the new capabilities.